### PR TITLE
docs - pxf config refactor, misc edits

### PR DESCRIPTION
**Pull request by lisakowen**
in this PR:
- remove unzip requirement, install openjdk-devel pkg
- hadoop config moved from /etc/xxx/conf to $PXF_CONF/servers/default - new steps in hadoop config and upgrade procedures
- new $PXF_CONF/templates directory
- corrected a few topic titles
- removed a step in pxf kerberos docs to copy hadoop config - the hadoop config should already be on the segment host
- misc edits

doc review site links:
- http://docs-lisa-pxfusrcfg2.cfapps.io/600/pxf/cfginitstart_pxf.html
- http://docs-lisa-pxfusrcfg2.cfapps.io/600/pxf/client_instcfg.html
- http://docs-lisa-pxfusrcfg2.cfapps.io/600/pxf/upgrade_pxf.html

### CODE REVIEW

File: gpdb-doc/markdown/pxf/client_instcfg.html.md.erb [hunk 1]
```diff
@@ -36,19 +31,20 @@ You will use the `gpssh` and `gpscp` utilities where possible to run a command o
     1. Copy the `core-site.xml`, `hdfs-site.xml`, `mapred-site.xml`, and `yarn-site.xml` Hadoop configuration files from your Hadoop cluster NameNode host to the current host using your tool of choice. For example, this command uses `scp` to copy the files:
 
         ``` shell
+        gpadmin@gpmaster$ cd $PXF_CONF/server/default
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/client_instcfg.html.md.erb]: This step uses $PXF_CONF/server/default for storing the files before copy, but the remaining steps use $PXF_CONF/server**s**/default as the destination.

File: gpdb-doc/markdown/pxf/pxfuserimpers.html.md.erb [hunk 2]
```diff
@@ -67,7 +67,7 @@ When PXF user personation is enabled (the default), you must configure the Hadoo
     ```
 4. After changing `core-site.xml`, you must restart Hadoop for your changes to take effect.
 
-5. Copy the updated `core-site.xml` file to the Hadoop configuration directory `/etc/hadoop/conf` on each Greenplum Database segment host.
+5. Copy the updated `core-site.xml` file to the PXF Hadoop configuration directory `$PXF_CONF/servers/default` on the master and each Greenplum Database segment host.
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/pxfuserimpers.html.md.erb]: and each -> and on each

File: gpdb-doc/markdown/pxf/sdk/deploy_conn.html.md.erb [hunk 3]
```diff
@@ -17,7 +17,7 @@ The PXF agent determines certain runtime dependencies from configuration files f
 
 The `$GPHOME/pxf/conf/pxf-private.classpath` configuration file identifies the runtime dependencies for the PXF agent itself. This file also identifies the runtime dependencies for the PXF built-in HDFS, Hive, and HBase connectors.
 
-The Greenplum Database administrator registers a third-party connector and its runtime dependencies by copying the files to the `$PXF_CONF/lib` directory.
+The Greenplum Database administrator registers a third-party connector and its runtime dependencies by copying the files to the `$PXF_CONF/lib` directory, and restarting the PXF agent on each segment host.
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/sdk/deploy_conn.html.md.erb]: , and -> , and then

File: gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb [hunk 4]
```diff
@@ -8,7 +8,7 @@ The PXF upgrade procedure describes how to upgrade PXF in your Greenplum Databas
 
 Most PXF installations do not require modifications to PXF configuration files and should experience a seamless upgrade.
 
-**Note**: Starting in Greenplum Database version 5.12.0, PXF no longer requires a Hadoop client installation. PXF now bundles all of the JAR files on which it depends, and loads these JARs at runtime. PXF still requires that the Hadoop configuration files reside in `/etc/<hadoop-service>/conf`.
+**Note**: Starting in Greenplum Database version 5.12.0, PXF no longer requires a Hadoop client installation. PXF now bundles all of the JAR files on which it depends, and loads these JARs at runtime.
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb]: 5.12.0 -> 5.12

File: gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb [hunk 5]
```diff
@@ -86,7 +86,8 @@ After you upgrade to the new version of Greenplum Database, perform the followin
...
     5. If you added additional JAR files to your *PXF.from* installation, copy them to `$PXF_CONF/lib` on all segment hosts.
-    5. Starting in Greenplum Database version 5.14, the default Kerberos keytab file location for PXF is `$PXF_CONF/keytabs`. If you previously configured PXF for secure HDFS and the PXF keytab file is located in a *PXF.from* installation directory (for example, `$GPHOME/pxf/conf`), consider relocating the keytab file to `$PXF_CONF/keytabs`. Alternatively, you can update the `PXF_KEYTAB` property setting in the `$PXF_CONF/conf/pxf-env.sh` file to reference a different keytab file directory. Be sure to propagate any updated files to each segment host.
+    5. Starting in Greenplum Database version 5.14, PXF requires that the Hadoop configuration files reside in the `$PXF_CONF/servers/default` directory. If you configured PXF Hadoop connectors in your *PXF.from* installation, copy the Hadoop configuration files in `/etc/<hadoop_service>/conf` to `$PXF_CONF/servers/default` on the Greenplum Database master and all segment hosts.
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb]: and all -> and on all

File: gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb [hunk 6]
```diff
@@ -86,7 +86,8 @@ After you upgrade to the new version of Greenplum Database, perform the followin
...
+    5. Starting in Greenplum Database version 5.14, the default Kerberos keytab file location for PXF is `$PXF_CONF/keytabs`. If you previously configured PXF for secure HDFS and the PXF keytab file is located in a *PXF.from* installation directory (for example, `$GPHOME/pxf/conf`), consider relocating the keytab file to `$PXF_CONF/keytabs`. Alternatively, you can update the `PXF_KEYTAB` property setting in the `$PXF_CONF/conf/pxf-env.sh` file to reference a different keytab file system path. Be sure to propagate any updated files to each segment host.
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb]: , you can update -> , update
a different keytab file system path -> your keytab file

(I'm assuming they need to either put the file in the correct location or update PXF_KEYTAB to point to it.)

File: gpdb-doc/markdown/pxf/client_instcfg.html.md.erb [hunk 7]
```diff
@@ -4,18 +4,13 @@ title: Configuring PXF Hadoop Connectors (Optional)
 
 PXF is compatible with Cloudera, Hortonworks Data Platform, and generic Apache Hadoop distributions. This topic describes how configure the PXF Hadoop, Hive, and HBase connectors.
 
-*If you do not plan to use the Hadoop-related PXF connectors, or you already have Hadoop client installations on your Greenplum Database segments hosts, you need not perform this procedure.*
+*If you do not plan to use the Hadoop-related PXF connectors, you need not perform this procedure.*
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/client_instcfg.html.md.erb]: If you do not want ..., then you do not need to perform ...

**Comment by lisakowen** [comment]
thank you for reviewing, @dyozie . good catch on the server->servers.

**Comment by lisakowen** [comment]
@lavjain - this PR should reflect the config refactor.  please review when you get a chance.

**Comment by lavjain** [comment]
@lisakowen Everything looks good. I would have preferred the PXF_CONF location to be `/etc/pxf/conf` instead of `/etc/pxf/usrcfg`

**Pull request merged**